research/blueprint_algo/average_strategy_short_deck.py

```diff
@@ -1,67 +1,75 @@
-import joblib
-from os import listdir
-from os.path import isfile, join
 import collections
+import glob
+import os
 from typing import Dict
 
+import click
+import joblib
+from tqdm import tqdm
 
-# TODO: I'm just rigging this to have ACTIONS as "fold", "call", and "raise", but
-#  we'll need to develop a way to load up the state given an infoset as retrieved
-#  from the strategies
-def calculate_strategy(
-    regret: Dict[str, Dict[str, float]],
-    sigma: Dict[int, Dict[str, Dict[str, float]]],
-    I: str,
-):
-    """
-
-    :param regret: dictionary of regrets, I is key, then each action at I, with values being regret
-    :param sigma: dictionary of strategy updated by regret, iteration is key, then I is key, then each action with prob
-    :param I:
-    :return: doesn't return anything, just updates sigma
-    """
-    rsum = sum([max(x, 0) for x in regret[I].values()])
-    ACTIONS = regret[I].keys()  # TODO: this is hacky, might be a better way
-    for a in ACTIONS:
-        if rsum > 0:
-            sigma[I][a] = max(regret[I][a], 0) / rsum
-        else:
-            sigma[I][a] = 1 / len(ACTIONS)
-    return sigma
 
+def calculate_strategy(this_info_sets_regret: Dict[str, float]) -> Dict[str, float]:
+    """Calculate the strategy based on the current information sets regret."""
+    # TODO: Could we instanciate a state object from an info set?
+    actions = this_info_sets_regret.keys()
+    regret_sum = sum([max(regret, 0) for regret in this_info_sets_regret.values()])
+    if regret_sum > 0:
+        strategy: Dict[str, float] = {
+            action: max(this_info_sets_regret[action], 0) / regret_sum
+            for action in actions
+        }
+    else:
```

Comment:
I forget, but I'm wondering if this means for each dump we default the strategy to 1 / state.legal_actions for each action. 

If we use that value to average, it may be more inaccurate than if we only averaged the found info sets across other found info sets. I'll take a look at this when I get to the PR but in case you have any thoughts